api/barriers/models.py

```diff
@@ -1395,9 +1395,23 @@ def sector_filter(self, queryset, name, value: List[str]):
         custom filter for multi-select filtering of Sectors field,
         which is ArrayField
         """
-        return queryset.filter(
-            Q(all_sectors=True) | Q(main_sector__in=value) | Q(sectors__overlap=value)
-        ).distinct()
+
+        ignore_all_sectors = self.data.get("ignore_all_sectors", False)
+        only_main_sector = self.data.get("only_main_sector", "no")
+
+        # If we're only concerned with main sectors and there's no provided value,
+        # we can directly check if main_sector is not null
+        if only_main_sector == "yes" and not value:
+            return queryset.filter(main_sector__isnull=False).distinct()
+
+        # Define base query based on provided options
+        base_query = Q(all_sectors=not ignore_all_sectors) | Q(main_sector__in=value)
+
+        if only_main_sector == "yes":
+            return queryset.filter(base_query).distinct()
+
+        # Add overlap condition for sectors
+        return queryset.filter(base_query | Q(sectors__overlap=value)).distinct()
```

Comment:
I think we can remove the .distinct() call here as Feroze's ticket should've resolved these duplicates
```suggestion
        return queryset.filter(base_query | Q(sectors__overlap=value))
```